msu/utils/serialization.nut

```diff
@@ -0,0 +1,132 @@
+::MSU.Serialization <- {
+	DataType = ::MSU.Class.Enum([
+		"None",
+		"Unknown",
+		"Null",
+		"Bool",
+		"String",
+		"U8",
+		"U16",
+		"U32",
+		"I8",
+		"I16",
+		"I32",
+		"F32",
+		"Table",
+		"Array",
+		"SerializationData"
+	]),
```

Comment:
This is still currently a problem, should maybe comment here that this should not be extended. This will currently break if extended by multiple mods because there is no ordering guarantee, the solution is to serialize and deserialize this enum alongside any data (probably into metadata). I assume we won't be doing this so instead comment here and in doc that this shouldn't be extended

Reply:
Based on our discussion we decided to postpone adding the ability to extend this enum to a future MSU release. For now, we will document it that modders must not extend it.